test/model_simulation/simulate_ODEs.jl

```diff
@@ -99,64 +116,88 @@ let
         du[2] = k2 * log(12 + X) * X - k3 * log(3 + Y) * Y
         du[3] = k3 * log(3 + Y) * Y - log(5, 6 + k4) * Z
     end
-    push!(identical_networks_1, reaction_networks_weird[2] => real_functions_5)
+    push!(catalyst_networks, reaction_networks_weird[2])
+    push!(manual_networks, real_functions_5)
+    push!(u0_syms, [:X, :Y, :Z])
+    push!(ps_syms, [:k1, :k2, :k3, :k4])
 
-    for (i, networks) in enumerate(identical_networks_1)
+    for (rn_catalyst, rn_manual, u0_sym, ps_sym) in zip(catalyst_networks, manual_networks, u0_syms, ps_syms)
         for factor in [1e-2, 1e-1, 1e0, 1e1]
-            u0 = factor * rand(rng, length(get_unknowns(networks[1])))
-            p = factor * rand(rng, length(get_ps(networks[1])))
-            (i == 3) && (p = min.(round.(p) .+ 1, 10))                      #If parameter in exponent, want to avoid possibility of (-small u)^(decimal). Also avoid large exponents.
-            prob1 = ODEProblem(networks[1], u0, (0.0, 10000.0), p)
-            sol1 = solve(prob1, Rosenbrock23(), saveat = 1.0)
-            prob2 = ODEProblem(networks[2], u0, (0.0, 10000.0), p)
-            sol2 = solve(prob2, Rosenbrock23(), saveat = 1.0)
-            @test all(abs.(hcat((sol1.u .- sol2.u)...)) .< 1e-7)
+            # Set input values.
+            u0_1 = rnd_u0(rn_catalyst, rng; factor = factor)
+            ps_1 = rnd_ps(rn_catalyst, rng; factor = factor)
+            (nameof(rn_catalyst) == :rnh2) && (p_1 = rnd_ps_Int64(rn_catalyst, rng))
+            u0_2 = map_to_vec(u0_1, u0_sym)
+            ps_2 = map_to_vec(ps_1, ps_sym)
+
+            # Check drift functions.
+            dt = zeros(length(u0_1))
+            rn_manual(dt, u0_2, ps_2, 0.0)
+            @test dt ≈ f_eval(rn_catalyst, u0_1, ps_1, 0.0)
+
+            # Compares that simulations are identical. 
+            oprob_1 = ODEProblem(rn_catalyst, u0_1, (0.0, 100.0), ps_1)
+            oprob_2 = ODEProblem(rn_manual, u0_2, (0.0, 100.0), ps_2)
+            sol1 = solve(oprob_1, Rosenbrock23())
+            sol2 = solve(oprob_2, Rosenbrock23())
+            @test sol1[u0_sym] ≈ sol2.u
```

Comment:
Indexes the solution simualted with Catalyst with the symbol vector showing the order the species appears in in the manual functions.